cli/tests/test_alias.rs

```diff
@@ -338,3 +343,88 @@ fn test_alias_in_repo_config() {
     aliases.l=["log", "-r@", "--no-graph", "-T\"user alias\\n\""]
     "###);
 }
+
+#[test]
+fn test_alias_help() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    test_env.add_config(r#"aliases.b = ["log", "-r", "@", "-T", "branches"]"#);
+    test_env.add_config(r#"aliases.s = ["status"]"#);
+    test_env.add_config(r#"aliases.f = ["git", "fetch"]"#); // test nested subcommand
+    test_env.add_config(r#"aliases.b2 = ["b", "--no-graph"]"#); // test recursive subcommand
+    test_env.add_config(r#"aliases.empty = []"#);
+    test_env.add_config(r#"aliases.option-only = ["--no-pager"]"#);
+    test_env.add_config(r#"aliases.bad = ["this-command-does-not-exist"]"#);
+
+    let stdout = test_env.jj_cmd_success(&repo_path, &["help", "b"]);
+    insta::assert_snapshot!(stdout.lines().take(3).join("\n"), @r###"
+    Alias for ["log", "-r", "@", "-T", "branches"]
```

Comment:
This isn't particularly human-readable.

How would you feel about `alias for "jj log -r @ -T branches"` or something to that extend. I think that would be nicer for default aliases in particular. A user just starting on jj who runs `jj help checkout` might not quite understand `alias for ["new"]`, but I imagine they'd understand `alias for "jj new"` perfectly.

We might need to use docs.rs/shlex/latest/shlex/fn.try_join.html to ensure that we don't print out wierd stuff when the user writes an alias as `["command", "with spaces", "or", "with_function_call()"]`

Reply:
We should also use that library here:
https://github.com/martinvonz/jj/blob/462c19736a9b3293b4cc9957404504718e19ef84/cli/src/cli_util.rs#L1464-L1465

Reply:
done. i haven't modified the existing code in `start_repo_transaction` because frankly i'm not sure how it works and i've already spent quite a while on this PR; i'd prefer to open a follow-up issue instead.